flang/test/Transforms/debug-ref-type.fir

```diff
@@ -0,0 +1,10 @@
+// RUN: fir-opt --add-debug-info --mlir-print-debuginfo %s | FileCheck %s
+
```

Comment:
Could you also verify the handling of `!fir.ref<none>`?

This will for instance be generated for the signature of `foo` after the abtsract-result pass:

```
subroutine test() 
  use iso_c_binding, only : c_ptr
  interface
  function foo() bind(c) 
    import :: c_ptr
    type(c_ptr) foo
  end function
  end interface
  call bar(foo())
end subroutine
```

```
func.func private @foo() -> !fir.ref<none> attributes {fir.bindc_name = "foo", fir.proc_attrs = #fir.proc_attrs<bind_c>}
```

Reply:
Thanks for your review. Currently `mlir::NoneType` is not handled and a place holder integer type gets generated for it. I have a small PR ready to add its support and I will add the `!fir.ref<none>` check in that.